File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()){
            return false;
        }
        map<char, int> count;
        for(int i = 0; i < s.length(); i++){
            count[s[i]] ++;
        //}
       // for(int i = 0; i < t.length(); i++){
            count[t[i]] --;
        }
        for(int i = 0; i < s.length(); i++){
            if(count[s[i]] != 0){
                return false;
            }
        }

        return true;
    }
};

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if head is None or head.next is None and n != 0:
            return None  
        
        last = head
        count_last = 1
        while last.next:
            last = last.next
            count_last += 1
        
        count = 1
        prev = head
        temp = None
        while prev:
            if count == count_last - (n-1):
                temp.next = prev.next
                return head
            
            temp = prev
            prev = prev.next
            count += 1